app/src/main/cpp/skyline/services/am/storage/IStorageAccessor.cpp

```diff
@@ -6,36 +6,40 @@
 #include "IStorageAccessor.h"
 
 namespace skyline::service::am {
-    IStorageAccessor::IStorageAccessor(const DeviceState &state, ServiceManager &manager, std::shared_ptr<IStorage> parent) : parent(parent), BaseService(state, manager, Service::am_IStorageAccessor, "am:IStorageAccessor", {
+    IStorageAccessor::IStorageAccessor(const DeviceState &state, ServiceManager &manager, std::shared_ptr<IStorage> parent) : parent(parent), BaseService(state, manager, {
         {0x0, SFUNC(IStorageAccessor::GetSize)},
         {0xA, SFUNC(IStorageAccessor::Write)},
         {0xB, SFUNC(IStorageAccessor::Read)}
     }) {}
 
-    void IStorageAccessor::GetSize(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+    Result IStorageAccessor::GetSize(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
         response.Push<i64>(parent->content.size());
+        return {};
     }
 
-    void IStorageAccessor::Write(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+    Result IStorageAccessor::Write(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
         auto offset = request.Pop<i64>();
-        auto size = request.inputBuf.at(0).size;
+        auto size = std::min(static_cast<i64>(request.inputBuf.at(0).size), static_cast<i64>(parent->content.size()) - offset);
```

Comment:
Why is the size an `i64` rather than `u64`? also replace `size > 0` with `size` in the conditional below with that change.

Reply:
Because of the offset potentially being larger than size

Reply:
Actually this kinda needs to be signed, it's the neatest way of doing it.

Reply:
the behaviour would be different